app/ht_indexer/src/ht_queue_service/queue_producer.py

```diff
@@ -1,70 +1,124 @@
+        self.channel = self.channel_creator.get_channel()
+
+
+        # Object to create the queue and manage its setup and attributes
+        self.queue_manager = QueueManager(self.queue_config)
+
+        # Ensure the queue is ready when the producer is initialized
+        if not self.queue_manager.is_ready(self.channel):
+            logger.warning("Queue setup not ready. Initializing channel and setup.")
+            self.queue_reconnect()
+
+    def queue_reconnect(self) -> None:
+        """
+        Re-establish the RabbitMQ connection and channel and set up the queue infrastructure.
+        This should be called when the connection or channel is closed unexpectedly.
+
+        :return: None
+        """
+        logger.info("Reconnecting to RabbitMQ...")
+
+        # Create a new channel
+        new_channel = self.channel_creator.get_channel()
 
+        # Replace the old channel with the new one
+        self.channel = new_channel
         try:
-            self.dlq_conn = QueueConnectionDeadLetter(user, password, self.host, self.queue_name, batch_size)
-        except Exception as e:
-            logger.error(
-                f"Failed to create dead-letter queue connection for {self.queue_name}: {e}"
-            )
+            # Set up the queue again if needed
+            self.queue_manager.set_up_queue(self.channel)
+        except Exception as err:
+            logger.error(f"Failed to reinitialize queue setup: {err}", exc_info=True)
             raise
 
-    def publish_messages(self, queue_message: dict) -> None:
-        # TODO Check if make sent to close the connection after publishing each message
+    def publish_messages(self, queue_message: Dict[str, Any]) -> None:
 
+        # TODO: Define a dataclass for the queue message to ensure type safety and validation
+        # TODO: Define a retry mechanism with exponential backoff for publishing messages
+
+        """
+        Publish the message. If the channel is closed during publishing, it will attempt to reconnect and retry
+        publishing. The message is serialized to JSON format before publishing.
+        :param queue_message: The message to be published should be a dictionary.
+        :return: None
+
+        :raises: Exception if the message cannot be serialized or published.
+        :raises: pika.exceptions.ChannelClosed and pika.exceptions.ConnectionClosed if the channel is closed during publishing.
+        :raises: pika.exceptions.ConnectionClosed if the connection is closed during publishing.
+        :raises: pika.exceptions.StreamLostError if the stream is lost during publishing.
+        :raises: pika.exceptions.ChannelWrongStateError if the channel is in the wrong state during publishing.
+        :raises: Other exceptions that may occur during publishing.
+        """
+        message_id = get_queue_message_id(queue_message)
         try:
+            if not self.channel or self.channel.is_closed:
```

Comment:
Some of the error handling code regarding the queuing is rather verbose. I'm wondering if there might be some way to extract or re-use some of it across methods/classes? Not something to worry about now but something to consider in the future?